diff --git a/src/hnsw.c b/src/hnsw.c
index bfc92a1..4a17b71 100644
--- a/src/hnsw.c
+++ b/src/hnsw.c
@@ -90,7 +90,7 @@ HnswInit(void)
 					  HNSW_DEFAULT_EF_CONSTRUCTION, HNSW_MIN_EF_CONSTRUCTION, HNSW_MAX_EF_CONSTRUCTION, AccessExclusiveLock);
 
 	DefineCustomIntVariable("hnsw.ef_search", "Sets the size of the dynamic candidate list for search",
-							"Valid range is 1..1000.", &hnsw_ef_search,
+							"Valid range is 1..10000.", &hnsw_ef_search,
 							HNSW_DEFAULT_EF_SEARCH, HNSW_MIN_EF_SEARCH, HNSW_MAX_EF_SEARCH, PGC_USERSET, 0, NULL, NULL, NULL);
 
 	DefineCustomEnumVariable("hnsw.iterative_scan", "Sets the mode for iterative scans",
diff --git a/src/hnsw.h b/src/hnsw.h
index 841af0f..714500c 100644
--- a/src/hnsw.h
+++ b/src/hnsw.h
@@ -47,7 +47,7 @@ typedef Pointer Item;
 #define HNSW_MAX_EF_CONSTRUCTION		1000
 #define HNSW_DEFAULT_EF_SEARCH	40
 #define HNSW_MIN_EF_SEARCH		1
-#define HNSW_MAX_EF_SEARCH		1000
+#define HNSW_MAX_EF_SEARCH		10000
 
 /* Tuple types */
 #define HNSW_ELEMENT_TUPLE_TYPE  1
